stdlib/private/StdlibCollectionUnittest/CMakeLists.txt

```diff
@@ -1,11 +1,19 @@
+set(SWIFT_SDK_OVERLAY_LIBRARY_BUILD_TYPES)
+if(SWIFT_BUILD_DYNAMIC_SDK_OVERLAY)
+  list(APPEND SWIFT_SDK_OVERLAY_LIBRARY_BUILD_TYPES SHARED)
+endif()
+if(SWIFT_BUILD_STATIC_SDK_OVERLAY)
+  list(APPEND SWIFT_SDK_OVERLAY_LIBRARY_BUILD_TYPES STATIC)
+endif()
```

Comment:
Rather than write these seven lines of code in each of the CMakeLists.txt files in `stdlib/private`, I think you could write it once in `stdlib/private/CMakeLists.txt`, then use the variables in each of the files. That'll significantly reduce the size of this pull request!